tool/tsh/common/tsh_test.go

```diff
@@ -3555,22 +3555,28 @@ func mockSSOLogin(t *testing.T, authServer *auth.Server, user types.User) client
 	return func(ctx context.Context, connectorID string, priv *keys.PrivateKey, protocol string) (*auth.SSHLoginResponse, error) {
 		// generate certificates for our user
 		clusterName, err := authServer.GetClusterName()
-		require.NoError(t, err)
+		assert.NoError(t, err)
+
 		sshCert, tlsCert, err := authServer.GenerateUserTestCerts(auth.GenerateUserTestCertsRequest{
-			Key:            priv.MarshalSSHPublicKey(),
-			Username:       user.GetName(),
-			TTL:            time.Hour,
-			Compatibility:  constants.CertificateFormatStandard,
-			RouteToCluster: clusterName.GetClusterName(),
+			Key:                  priv.MarshalSSHPublicKey(),
+			Username:             user.GetName(),
+			TTL:                  time.Hour,
+			Compatibility:        constants.CertificateFormatStandard,
+			RouteToCluster:       clusterName.GetClusterName(),
+			AttestationStatement: priv.GetAttestationStatement(),
 		})
-		require.NoError(t, err)
+		if err != nil {
+			// return this error since it is essentially a login error
+			// and might be expected as a result in some tests.
+			return nil, trace.Wrap(err)
```

Comment:
the use of `testing.T` is good for test debugging, in case the test creator is missing some setup required for this method to work.

However, I agree and think we should just remove the `testing.T` from this call. It was causing a panic in my test since it expects an error from here. Outside of the debugging purposes, I don't think there is any negative to removing the `t`. Any issues should just be propagated through `tsh` to the test caller.

Reply:
37d328f789dda7a65b50e2937e91621ce52fef76

Might need to move this to a follow up PR?

Reply:
Up to you.